docs/dev_guide/contents/extending_fido.rst

```diff
@@ -0,0 +1,501 @@
+A "scraper" client inherits from `~sunpy.net.dataretriever.client.GenericClient` which provides helper methods for downloading from a list of URLs.
+If your web service provides a list of HTTP or FTP URLs that can easily be obtained from a search, this is probably the best approach.
+If the service you want to add requires making requests to an API with parameters for the search and getting a list of results in return, then you probably want to write a "full" client, which gives you full control.
+
+Before writing a new client, ensure you are familiar with how searches are specified by the `sunpy.net.attr` system including combining them with logical operations.
+
+
+.. _new_scraper_client:
+
+Writing a new "scraper" client
+==============================
+
+A "scraper" Fido client (also sometimes referred to as a "data retriever" client) is a Fido client which uses the URL `~sunpy.util.scraper.Scraper` to find files on remote servers.
+If the data provider you want to integrate does not provide a tree of files with predictable URLs then a "full" client is more likely to provide the functionality you need.
+
+A new "scraper" client inherits from `~sunpy.net.dataretriever.client.GenericClient` and requires a minimum of these three components:
+
+* A class method :meth:`~sunpy.net.dataretriever.client.GenericClient.register_values` which registers the "attrs" that are supported by the client.
+  It returns a dictionary where keys are the supported attrs and values are lists of tuples.
+  Each ``tuple`` contains the attr value and its description.
+* A class attribute ``baseurl``; this is a regex which is used to match all URLs supported by the client.
+* A class attribute ``pattern``; this is a template used to extract the metadata from URLs matched by ``baseurl``, the extraction uses  the :func:`~sunpy.extern.parse.parse` format.
+
+
+For a simple example of a scraper client, we can look at the implementation of `~.EVEClient` in sunpy, a version without docstrings is reproduced below:
+
+.. code-block:: python
+
+  class EVEClient(GenericClient):
+      baseurl = (r'http://lasp.colorado.edu/eve/data_access/evewebdata/quicklook/'
+                 r'L0CS/SpWx/%Y/%Y%m%d_EVE_L0CS_DIODES_1m.txt')
+      pattern = '{}/SpWx/{:4d}/{year:4d}{month:2d}{day:2d}_EVE_L{Level:1d}{}'
+
+      @classmethod
+      def register_values(cls):
+          from sunpy.net import attrs
+          adict = {attrs.Instrument: [('EVE', 'Extreme ultraviolet Variability Experiment, which is part of the NASA Solar Dynamics Observatory mission.')],
+                   attrs.Physobs: [('irradiance', 'the flux of radiant energy per unit area.')],
+                   attrs.Source: [('SDO', 'The Solar Dynamics Observatory.')],
+                   attrs.Provider: [('LASP', 'The Laboratory for Atmospheric and Space Physics.')],
+                   attrs.Level: [('0', 'EVE: The specific EVE client can only return Level 0C data. Any other number will use the VSO Client.')]}
```

Comment:
Is this really 'Level  0C' or just 'Level 0'

Reply:
Also might be worth mentioning these are converted to lowercase attributes on the attrs `a.Source.sdo`?

Reply:
> Is this really 'Level 0C' or just 'Level 0'

errr, :man_shrugging: I think this is probably outside the scope of this PR lol.

> Also might be worth mentioning these are converted to lowercase attributes on the attrs a.Source.sdo?

:+1: